### DISCUSSION TITLE
Display value based on user's current language

### DISCUSSION THREAD

**Discussion by jasonpughironbyte** [khoait/DCE.PCF#187] [Q&A] [answered]
First off, thank you for all of the hard work you have put into this control, it is great.  I was wondering if you could tell me if there is a way to configure PolyLookup to show the value from a field based on the user's currently selected language.  For example, if the user is currently viewing the application in English, bind PolyLookup to column ssc_nameenglish, if the user is viewing the application in French, bind PolyLookup to column ssc_namefrench.  I'm currently using the dynamic Fetch XML option:

```
<fetch>
  <entity name="ssc_issueoption">
    <attribute name="ssc_nameenglish" />   *Is there a way to bind to a different column based on user's current language*
    <filter>
      <condition attribute="ssc_issue" operator="eq" value="{{ ssc_issue.id }}" />
    </filter>
  </entity>
</fetch>
```

Thanks!

**Comment by khoait** [comment, marked as answer]
hi @jasonpughironbyte thanks for your interest in the control! your scenario is a perfect use case for [binding lookup config to a field](https://github.com/khoait/DCE.PCF/wiki/PolyLookup#binding-to-a-form-field). please have a look in the wiki.

basically, you'll set the `lookup view` property to a form field, then you'll need to write a form script to set the field value accordingly to user language. you can use any types of lookup view config or mix-and-match different types. to keep it simple, let's use system view name:
1. create 2 views: `PolyLookup Issues View (EN)` and `PolyLookup Issues View (FR)` with language specific columns in each view
2. create a hidden field on the form `new_polylookup_lookupview`
3. create an onload form script to
    - retrieve user language
    - check if user language is French, set `new_polylookup_lookupview` value to `PolyLookup Issues View (FR)`; 
    - otherwise, set `new_polylookup_lookupview` value to `PolyLookup Issues View (EN)` as default
4. optionally, [set submit mode](https://learn.microsoft.com/en-us/power-apps/developer/model-driven-apps/clientapi/reference/attributes/setsubmitmode?WT.mc_id=DX-MVP-5004717) of the field to `never` if you don't want to save the view name in the hidden field to database

hope this helps!

**Reply by jasonpughironbyte** [reply]
Hey @khoait, I implemented the suggestion you mentioned above, and it appears to be working properly for the most part, but I am seeing some strange behaviour.  What I am seeing is that everything works great before saving the record.  Once the record is saved initially, the selected values in PolyLookup revert to displaying the value from the Primary Name column.  I've implemented the following JavaScript to set the FetchXML:


```
function setIssueOptionFetchXMLStatement(executionContext)
{
	
	var formContext = executionContext.getFormContext();
   	var languageId = Xrm.Utility.getGlobalContext().userSettings.languageId;
		
	formContext.getAttribute("ssc_issueoptionview").setSubmitMode("never");
	
	if (languageId == 1036)
	{
		formContext.getAttribute("ssc_issueoptionview").setValue(
		
			"<fetch>" + 
			"	<entity name='ssc_issueoption'>" + 
			"		<attribute name='ssc_namefrench' />" + 
			"		<filter>" + 
			"			<condition attribute='ssc_issue' operator='eq' value='{{ ssc_issue.id }}' />" + 
			"		</filter>" + 
			"	</entity>" + 
			"</fetch>" 
		
		);		
	}	
	else
	{
		formContext.getAttribute("ssc_issueoptionview").setValue(
		
			"<fetch>" + 
			"	<entity name='ssc_issueoption'>" + 
			"		<attribute name='ssc_nameenglish' />" + 
			"		<filter>" + 
			"			<condition attribute='ssc_issue' operator='eq' value='{{ ssc_issue.id }}' />" + 
			"		</filter>" + 
			"	</entity>" + 
			"</fetch>" 

		
		);		
	}		
}

```

Before the form is saved, this appears to work as expected:

![image](https://github.com/user-attachments/assets/70e58901-7c89-4711-b6f7-8ade0a127da6)

However, after the save, it reverts the display field back to the Primary Name field:

![image](https://github.com/user-attachments/assets/781d9fe5-4f77-4666-bed5-d426af251973)



Any idea why it would go back to default name field (ssc_name)?

**Reply by jasonpughironbyte** [reply]
When I Monitor the application, I can see that the following FetchXML statement is being sent after opening a saved record:

```
<fetch page="1" count="50">	
  <entity name="ssc_issueoption">		
     <attribute name="ssc_nameenglish"/>		
       <filter>			
          <condition attribute="ssc_issue" operator="eq" value="CB3FFACB-E675-EF11-AC21-0022483CA08E"/>		
       </filter>	

      <attribute name="ssc_issueoptionid"/>  // These two attributes have been added to the FetchXML statement
      <attribute name="ssc_name"/>

    </entity>
</fetch>
```

**Reply by khoait** [reply]
hi @jasonpughironbyte, good findings! they are primary id and primary name attributes. currently the control only display primary name column for selected items. there is an open feature to display custom text with dynamic templating for selected item display text here: https://github.com/khoait/DCE.PCF/issues/170. I'll work on it and will include it in the next release.

**Reply by jasonpughironbyte** [reply]
@khoait Ok, thank you, I thought maybe I was doing something wrong.  Thanks again for your help!

**Reply by khoait** [reply]
@jasonpughironbyte I just realized your sponsorship! really appreciate it! I couldn't find a way to contact you, so I just put it here. do you mind dropping me an email so we can communicate better? 
Thanks, Khoa

**Reply by khoait** [reply]
@jasonpughironbyte happy to share that this is now possible in the new release v1.1.1.0.
 the control is now using the first column in the fetchxml/view as display text by default. you can update the fetchxml according to user's language.
alternatively, you can use the new property `Selected Item Template` with the mustache syntax to format the display text as needed. see more here: https://github.com/khoait/DCE.PCF/wiki/PolyLookup#selected-item-text
